### PR TITLE
Add `doc_path`s to 1.56 and 1.57 features

### PR DESCRIPTION
**Pull request by robjtede**
Also removes `try_reserve_exact` features for HashMap and HashSet which will not exist [according to the tracking issue](https://github.com/rust-lang/rust/issues/48043).

### CODE REVIEW

File: data/1.56/extend_for_tuple.toml [hunk 1]
```diff
@@ -1,3 +1,4 @@
 title = "`Extend<A, B>` implementation for `(Extend<A>, Extend<B>)`"
 flag = "extend_for_tuple"
 impl_pr_id = 85835
+doc_path = "std/iter/trait.Extend.html#impl-Extend<(A%2C B)>"
```
Comment (jplatte) [inline data/1.56/extend_for_tuple.toml:4]: I think this space should be escaped for broader browser compatibility.

File: data/1.56/std_collections_from_array.toml [hunk 2]
```diff
@@ -10,3 +10,4 @@ items = [
     "impl From<T; N> for LinkedList<T>",
     "impl From<T; N> for VecDeque<T>",
 ]
+doc_path = "std/collections/struct.VecDeque.html#impl-From<[T%3B N]>"
```
Comment (jplatte) [inline data/1.56/std_collections_from_array.toml:13]: Same here.

File: data/1.57/vec_try_reserve.toml [hunk 3]
```diff
@@ -3,4 +3,4 @@ flag = "try_reserve"
 impl_pr_id = 48648
 tracking_issue_id = 48043
 stabilization_pr_id = 87993
-doc_path = ""
+doc_path = "std/vec/struct.Vec.html#method.try_reserve"
```
Comment (jplatte) [inline data/1.57/vec_try_reserve.toml:6]: Should ideally be part of the previous commit, but not a huge deal.

Comment (robjtede) [reply data/1.57/vec_try_reserve.toml:6]: goodness, you _are_ a thorough reviewer :)

**Comment by jplatte** [comment]
Thanks!

**Pull request merged**